src/js/modules/rpc/service.ts

```diff
@@ -45,30 +43,66 @@ const validateOrCreateScaffolding = (directoryPath: string): Promise<void> => {
   return Promise.all(validations).then(() => null);
 };
 
+export const closeProcess = (e: Error): Promise<void> => {
+  return LogService.close().then(() => {
+    fs.writeFile(
+      LogService.getPath(),
+      `Error: ${e.message}`,
+      { flag: "+a" },
+      (err) => {
+        if (err) {
+          console.error(
+            "failing on write exception on " +
+              LogService.getPath() +
+              " Error: " +
+              err.message
+          );
+        }
+        console.log("Closing");
+        process.exit(1);
+      }
+    );
+  });
+};
+
 function main() {
   // read config file path argument
-  logger.info("Starting redpanda wasm service...");
   const configPathArg = process.argv.splice(2)[0];
-  const defaultConfigPath = "/var/lib/redpanda/conf/redpanda.yaml";
+  const defaultConfigPath = "/etc/redpanda/redpanda.yaml";
   const defaultCoprocessorPath = "/var/lib/redpanda/coprocessor";
+  const defaultLogFile = "/var/lib/redpanda/coprocessor/logs/wasm";
   // resolve config path or assign default value
   const configPath = configPathArg ? resolve(configPathArg) : defaultConfigPath;
-  logger.info(`Using redpanda configuration path: ${configPath}`);
 
-  readConfigFile(configPath).then((config) => {
-    const port = config?.redpanda?.coproc_supervisor_server || 43189;
-    const path = config?.coproc_engine?.path || defaultCoprocessorPath;
-    logger.info(`Using root scaffolding path: ${path}`);
-    validateOrCreateScaffolding(path).then(() => {
-      const service = new ProcessBatchServer(
-        join(path, "active"),
-        join(path, "inactive"),
-        join(path, "submit")
-      );
-      logger.info(`Starting redpanda wasm service on port: ${port}`);
-      service.listen(port);
-    });
-  });
+  readConfigFile(configPath)
+    .then((config) => {
+      const port = config?.redpanda?.coproc_supervisor_server || 43189;
+      const path = config?.coproc_engine?.path || defaultCoprocessorPath;
+      const logFilePath = config?.coproc_engine?.logFilePath || defaultLogFile;
+      LogService.setPath(logFilePath);
+      const logger = LogService.createLogger("service");
+      logger.info("Starting redpanda wasm service...");
+      logger.info(`Reading from config file: ${configPath}`);
+      logger.info(`Using root scaffolding path: ${path}`);
+      validateOrCreateScaffolding(path)
+        .then(() => {
+          const service = new ProcessBatchServer(
+            join(path, "active"),
+            join(path, "inactive"),
+            join(path, "submit")
+          );
+          process.on("SIGINT", function () {
```

Comment:
Why don't we use the unhandled exception handler? If we do it would just be one place to call closeProcess() instead of 3 points you have, plus this SIGINT signal handler

Reply:
because winston doesn't handle exception into promise (`UnhandledPromiseRejectionWarning`) for that reason that Winston attribute doesn't work for us.

Reply:
https://github.com/winstonjs/winston/issues/1617#

Reply:
Looks like they added something called a 'RejectionHandler' but someone said it isn't working as of July 9th 2019. Can you test and see if thats still the case? Should be easy to create a little script that throws within a promise to see if the callback is called

Reply:
Yes, I tested but it doesn't work.